docs/source/en/package_reference/dataclasses.md

```diff
@@ -0,0 +1,220 @@
+            raise ValueError(f"vocab_size ({self.vocab_size}) must be greater than hidden_size ({self.hidden_size})")
+```
+
+Fields are validated during initialization:
+
+```python
+config = Config(model_type="bert", hidden_size=24)   # Valid
+config = Config(model_type="bert", hidden_size=-1)   # Raises StrictDataclassFieldValidationError
+```
+
+Consistency between fields is also validated during initialization (class-wise validation):
+
+```python
+# `vocab_size` too small compared to `hidden_size`
+config = Config(model_type="bert", hidden_size=32, vocab_size=16)   # Raises StrictDataclassClassValidationError
+```
+
+Fields are also validated during assignment:
+
+```python
+config.hidden_size = 512  # Valid
+config.hidden_size = -1   # Raises StrictDataclassFieldValidationError
+```
+
+To re-run class-wide validation after assignment, you must call `.validate` explicitly:
+
+```python
+config.validate()  # Runs all class validators
+```
+
+### Custom Validators
+
+You can attach multiple custom validators to fields using [`validated_field`]. A validator is a callable that takes a single argument and raises an exception if the value is invalid.
+
+```python
+from dataclasses import dataclass
+from huggingface_hub.dataclasses import strict, validated_field
+
+def multiple_of_64(value: int):
+    if value % 64 != 0:
+        raise ValueError(f"Value must be a multiple of 64, got {value}")
+
+@strict
+@dataclass
+class Config:
+    hidden_size: int = validated_field(validator=[positive_int, multiple_of_64])
+```
+
+In this example, both validators are applied to the `hidden_size` field.
+
+### Additional Keyword Arguments
+
+By default, strict dataclasses only accept fields defined in the class. You can allow additional keyword arguments by setting `accept_kwargs=True` in the `@strict` decorator.
+
+```python
+from dataclasses import dataclass
+from huggingface_hub.dataclasses import strict
+
+@strict(accept_kwargs=True)
+@dataclass
+class ConfigWithKwargs:
+    model_type: str
+    vocab_size: int = 16
+
+config = ConfigWithKwargs(model_type="bert", vocab_size=30000, extra_field="extra_value")
+print(config)  # ConfigWithKwargs(model_type='bert', vocab_size=30000, *extra_field='extra_value')
+```
+
+Additional keyword arguments appear in the string representation of the dataclass but are prefixed with `*` to highlight that they are not validated.
+
+### Integration with Type Hints
+
+Strict dataclasses respect type hints and validate them automatically. For example:
+
+```python
+from typing import List
+from dataclasses import dataclass
+from huggingface_hub.dataclasses import strict
+
+@strict
+@dataclass
+class Config:
+    layers: List[int]
+
+config = Config(layers=[64, 128])  # Valid
+config = Config(layers="not_a_list")  # Raises StrictDataclassFieldValidationError
+```
+
+Supported types include:
+- Any
+- Union
+- Optional
+- Literal
+- List
+- Dict
+- Tuple
+- Set
+
+And any combination of these types.
```

Comment:
Suggestion: mention that custom type validator can still be added through custom validators. 

e.g.
```suggestion
And any combination of these types. If validation for a different type is desired, it can be done through a custom validator.
```